landavailability/api/models.py

```diff
@@ -40,6 +42,23 @@ class BusStop(models.Model):
     road = models.CharField(max_length=255, blank=True, null=True)
     nptg_code = models.CharField(max_length=255, blank=True, null=True)
 
+    def update_close_locations(self, distance=1000):
+        locations = Location.objects.filter(
+            geom__dwithin=(self.point, D(m=distance)))
+
+        for location in locations:
+            distance = location.geom.distance(self.point)
+
+            if location.nearest_busstop:
+                if distance > location.geom.distance(
+                        location.nearest_busstop.point):
+                    continue
+
+            location.nearest_busstop = self
+            # FIXME: distance is espressed in a linear way
```

Comment:
According to this https://docs.djangoproject.com/en/1.10/ref/contrib/gis/geos/#django.contrib.gis.geos.GEOSGeometry.distance
**"GEOS distance calculations are linear – in other words, GEOS does not perform a spherical calculation even if the SRID specifies a geographic coordinate system"** but it doesn't say how to calculate the proper distance. The workaround I've used is to implement the method **refresh_busstop_distance** in Location model, what do you think about that? p.s: it needs to be tested properly yet.